awx/main/models/events.py

```diff
@@ -484,29 +487,45 @@ def _update_host_summary_from_stats(self, hostnames):
             if not self.job or not self.job.inventory:
                 logger.info('Event {} missing job or inventory, host summaries not updated'.format(self.pk))
                 return
-            qs = self.job.inventory.hosts.filter(name__in=hostnames)
             job = self.job
+
+            from awx.main.models import Host, JobHostSummary  # circular import
+            existing = Host.objects.filter(id__in=self.host_map.values()).values_list('id', flat=True)
```

Comment:
one point, I feel like `existing` should be a set. This is kind of the metric we care about scaling with, so, even though it's in-memory, seems like it's worth doing.

Also, this seems really inefficient / duplicated with the subsequent line

```python
            all_hosts = Host.objects.filter(
                pk__in=host_mapping.keys()
            ).only('id')
```

Starting out here, first thing, you could define `all_hosts`. Then from that, create `existing`.